### DISCUSSION TITLE
ESP32 compatibility

### DISCUSSION THREAD

**Discussion by bobwho69** [jblance/mpp-solar#159] [Q&A]
Hi jblance, I must say your code is working flawlessly on my pi setup all year and I am about to make the move to more automation and centralised intergration.  I see your micropython slave code (discussion not open there sorry) is already at a certain stage and possibly working, is there a chance it is compatible with the lan8720A library on a W32-ETH01?

For background I have 4x pip4048 inverters with my pi in an outdoor cabinet and a jkbms in a 20ft container all away from the house so wifi is not an option for me, but I have ethernet.  Currently using Influx and Grafana displayed on an old wall mounted tablet and using RaspController android app to send preset ssh commands based on what I see.  It works, but I want the next level of automation.

I would like to run a WT32-ETH01 to my jkbms (not yet datalogged) via rs485 and then eventually replace my pi with another WT32-ETH01 to move the pi inside the house.  and I'm also going to add Home Assistant to play with more IOT (and because my panasonic a/c has the worst cloud app ever) I also dont know if I will need to integrate Node Red into it all as well or not.

Thanks for any advice to set me on the right path...

edit:grammar

**Comment by syssi** [comment]
What about an ESPHome approach ;-)

https://github.com/syssi/esphome-jk-bms
https://github.com/syssi/esphome-pipsolar

**Reply by jblance** [reply]
@syssi wow - great work!
I started on something like this but failed to get the ESP to talk serial to the inverter
Now if you could make a version that just listens for commands on a mqtt topic and sends the results on another that would be perfect :-)

**Reply by bobwho69** [reply]
@syssi Thanks very much for this, I will already be implementing ESPHome for the AC so I will def give it a try!

**Comment by jblance** [comment]
Hi

The slave code on my repo isnt complete 
My preference is to make a really light weight component (probably using ESPHome as mentioned above) that listens on a mqtt topic for commands, passes the command to the inverter and then publishes the results on another mqtt topic - the commands and results would need to be encoded (eg something like `binascii.hexlify(full_command)`). 

This would allow a very simple ESP setup and allow for the logic of command creation and decoding to be done more centrally

I started on this, but failed to get the ESP talking serial to my inverter - but it looks like someone has done all the work for the PI30 protocol - though this would work for you

Re running as a service - raise an issue with what you have done, what works and what doesnt, the code does allow a fair amount of testing on the commandline to make sure things work before 'hiding' it as a service
One of the problems with the service is that there isnt a way to generate a date etc for commands that need that (yet)

**Reply by bobwho69** [reply]
Thanks for the heads up on the code, it's still out of my depth to develop, but I like your intentions and will check back some time :)

**Comment by JakobTewes** [comment]
@jblance, did you succeed with an ESPHome integration? Highly interested in this ;-)

**Reply by jblance** [reply]
no, there are some by @syssi and as far as i know they are feature complete and very good
